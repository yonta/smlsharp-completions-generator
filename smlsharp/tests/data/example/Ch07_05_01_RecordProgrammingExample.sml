fun moveX (p as {X:real,Vx:real,...}, t:real) = p # {X = X + Vx};
fun moveY (p as {Y:real,Vy:real,...}, t:real) = p # {Y = Y + Vy};
fun accelerateX (p as {Vx:real,...}, t:real) = p;
fun accelerateY (p as {Vy:real,...}, t:real) = p # {Vy = Vy + 9.8};

fun tic (p, t) =
let
  val p = accelerateX (p, t)
  val p = accelerateY (p, t)
  val p = moveX (p, t)
  val p = moveY (p, t)
in
  p
end;

fun accelerateX (p as {Vx:real,...}, t:real) = p # {Vx = Vx * 0.90};
