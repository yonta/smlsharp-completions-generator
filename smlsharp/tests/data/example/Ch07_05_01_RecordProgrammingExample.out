val moveX = fn : ['a#{Vx: real, X: real}. 'a * real -> 'a]

val moveY = fn : ['a#{Vy: real, Y: real}. 'a * real -> 'a]

val accelerateX = fn : ['a#{Vx: real}. 'a * real -> 'a]

val accelerateY = fn : ['a#{Vy: real}. 'a * real -> 'a]

val tic = fn : ['a#{Vx: real, Vy: real, X: real, Y: real}. 'a * real -> 'a]

val accelerateX = fn : ['a#{Vx: real}. 'a * real -> 'a]
